homeassistant/components/switch/fritzhome.py

```diff
@@ -0,0 +1,90 @@
+"""
+Support for AVM Fritz!Box fritzhome switch devices.
+
+For more details about this component, please refer to the documentation at
+http://home-assistant.io/components/switch.fritzhome/
+"""
+import logging
+from homeassistant.components.fritzhome import (
+    ATTR_AIN, ATTR_FW_VERSION, ATTR_ID, ATTR_MANUFACTURER, ATTR_PRODUCTNAME,
+    DOMAIN)
+from homeassistant.components.switch import (SwitchDevice)
+
+DEPENDENCIES = ['fritzhome']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Fritzhome switch platform."""
+    if DOMAIN not in hass.data:
+        return False
+
+    device_list = hass.data[DOMAIN]
+
+    devices = []
+    for device in device_list:
+        if device.has_switch:
+            devices.append(FritzhomeSwitch(hass, device))
+
+    add_devices(devices)
+
+
+class FritzhomeSwitch(SwitchDevice):
+    """The switch class for Fritzhome switches."""
+
+    def __init__(self, hass, device):
+        """Initialize the switch."""
+        self._device = device
+        self._state = None
+
+    @property
+    def available(self):
+        """Return if switch is available."""
+        return self._device.present
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._device.name
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        attr = {
+            ATTR_AIN: self._device.ain,
+            ATTR_FW_VERSION: self._device.fw_version,
```

Comment:
We should never include firmware version, manufacturer or product name in the device state attributes. Please remove.

Attributes should only contain info about the state that is relevant for automations.